accounts/store_sql.go

```diff
@@ -0,0 +1,704 @@
+			LastUpdated:        s.clock.Now().UTC(),
+			Label:              labelVal,
+			Alias:              alias,
+		})
+		if err != nil {
+			return fmt.Errorf("inserting account: %w", err)
+		}
+
+		account, err = getAndMarshalAccount(ctx, db, id)
+		if err != nil {
+			return fmt.Errorf("fetching account: %w", err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return account, nil
+}
+
+// getAndMarshalAccount retrieves the account with the given ID. If the account
+// cannot be found, then ErrAccNotFound is returned.
+func getAndMarshalAccount(ctx context.Context, db SQLQueries, id int64) (
+	*OffChainBalanceAccount, error) {
+
+	dbAcct, err := db.GetAccount(ctx, id)
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, ErrAccNotFound
+	} else if err != nil {
+		return nil, err
+	}
+
+	return marshalDBAccount(ctx, db, dbAcct)
+}
+
+func marshalDBAccount(ctx context.Context, db SQLQueries,
+	dbAcct sqlc.Account) (*OffChainBalanceAccount, error) {
+
+	alias, err := AccountIDFromInt64(dbAcct.Alias)
+	if err != nil {
+		return nil, err
+	}
+
+	account := &OffChainBalanceAccount{
+		ID:             alias,
+		Type:           AccountType(dbAcct.Type),
+		InitialBalance: lnwire.MilliSatoshi(dbAcct.InitialBalanceMsat),
+		CurrentBalance: dbAcct.CurrentBalanceMsat,
+		LastUpdate:     dbAcct.LastUpdated.UTC(),
+		ExpirationDate: dbAcct.Expiration.UTC(),
+		Invoices:       make(AccountInvoices),
+		Payments:       make(AccountPayments),
+		Label:          dbAcct.Label.String,
+	}
+
+	invoices, err := db.ListAccountInvoices(ctx, dbAcct.ID)
+	if err != nil {
+		return nil, err
+	}
+	for _, invoice := range invoices {
+		var hash lntypes.Hash
+		copy(hash[:], invoice.Hash)
+		account.Invoices[hash] = struct{}{}
+	}
+
+	payments, err := db.ListAccountPayments(ctx, dbAcct.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, payment := range payments {
+		var hash lntypes.Hash
+		copy(hash[:], payment.Hash)
+		account.Payments[hash] = &PaymentEntry{
+			Status:     lnrpc.Payment_PaymentStatus(payment.Status),
+			FullAmount: lnwire.MilliSatoshi(payment.FullAmountMsat),
+		}
+	}
+
+	return account, nil
+}
+
+// uniqueRandomAccountAlias generates a random account alias that is not already
+// in use. An account "alias" is a unique 8 byte identifier (which corresponds
+// to the AccountID type) that is used to identify accounts in the database. The
+// reason for using this alias in addition to the SQL auto-incremented ID is to
+// remain backwards compatible with the kvdb implementation of the DB which only
+// used the alias.
+func uniqueRandomAccountAlias(ctx context.Context, db SQLQueries) (int64,
+	error) {
+
+	var (
+		newAlias AccountID
+		numTries = 10
+	)
+	for numTries > 0 {
+		if _, err := rand.Read(newAlias[:]); err != nil {
+			return 0, err
+		}
+
+		newAliasID, err := newAlias.ToInt64()
+		if err != nil {
+			return 0, err
+		}
+
+		aliasInt, err := newAlias.ToInt64()
+		if err != nil {
+			return 0, err
+		}
```

Comment:
Both of these calls newAlias.ToInt64(), so seems unnecessary to create two variables.

Reply:
oversight 👍